src/mono/mono/metadata/mono-debug.c

```diff
@@ -1067,88 +1068,73 @@ mono_is_debugger_attached (void)
 	return is_attached;
 }
 
-/*
- * Bundles
- */
-
-typedef struct _BundledSymfile BundledSymfile;
-
-struct _BundledSymfile {
-	BundledSymfile *next;
-	const char *aname;
-	const mono_byte *raw_contents;
-	int size;
-};
-
-static BundledSymfile *bundled_symfiles = NULL;
-
 /**
  * mono_register_symfile_for_assembly:
+ * Dynamically allocates MonoBundledAssemblyResource to leverage
+ * preferred bundling api mono_bundled_resources_add.
  */
 void
 mono_register_symfile_for_assembly (const char *assembly_name, const mono_byte *raw_contents, int size)
 {
-	BundledSymfile *bsymfile;
-
-	bsymfile = g_new0 (BundledSymfile, 1);
-	bsymfile->aname = assembly_name;
-	bsymfile->raw_contents = raw_contents;
-	bsymfile->size = size;
-	bsymfile->next = bundled_symfiles;
-	bundled_symfiles = bsymfile;
-}
-
-static gboolean
-bsymfile_match (BundledSymfile *bsymfile, const char *assembly_name)
-{
-	if (!strcmp (bsymfile->aname, assembly_name))
-		return TRUE;
-#ifdef ENABLE_WEBCIL
-	const char *p = strstr (assembly_name, ".webcil");
-	/* if assembly_name ends with .webcil, check if aname matches, with a .dll extension instead */
-	if (p && *(p + strlen(".webcil")) == 0) {
-		size_t n = p - assembly_name;
-		if (!strncmp (bsymfile->aname, assembly_name, n)
-			&& !strcmp (bsymfile->aname + n, ".dll"))
-			return TRUE;
-	}
-	p = strstr (assembly_name, MONO_WEBCIL_IN_WASM_EXTENSION);
-	if (p && *(p + strlen(MONO_WEBCIL_IN_WASM_EXTENSION)) == 0) {
-		size_t n = p - assembly_name;
-		if (!strncmp (bsymfile->aname, assembly_name, n)
-			&& !strcmp (bsymfile->aname + n, ".dll"))
-			return TRUE;
-	}
-#endif
-	return FALSE;
+	mono_bundled_resources_add_assembly_symbol_resource (assembly_name, raw_contents, size, NULL, NULL);
 }
 
 static MonoDebugHandle *
 open_symfile_from_bundle (MonoImage *image)
 {
-	BundledSymfile *bsymfile;
+	MonoDebugHandle *handle = NULL;
+	const uint8_t *data = NULL;
+	uint32_t size = 0;
+	if (mono_bundled_resources_get_assembly_resource_values (image->module_name, NULL, NULL, &data, &size))
+		handle = mono_debug_open_image (image, data, size);
 
-	for (bsymfile = bundled_symfiles; bsymfile; bsymfile = bsymfile->next) {
-		if (!bsymfile_match (bsymfile, image->module_name))
-			continue;
-
-		return mono_debug_open_image (image, bsymfile->raw_contents, bsymfile->size);
+#ifdef ENABLE_WEBCIL
```

Comment:
Is this needed? Doesn't mono_bundled_resources_get_assembly_resource_values  take care of aliasing .webcil and MONO_WEBCIL_IN_WASM_EXTENSION -> .dll as part of search logic?

Reply:
Good point, removing this guard